### PR TITLE
docs: added vite docs

### LINKED ISSUE #1885: Vite build error
### Qwik Version

0.10.0 or greater

### Operating System (or Browser)

Stackblitz container

### Node Version (if applicable)

v16.14.2

### Which component is affected?

Qwik Rollup / Vite plugin

### Expected Behaviour

The build should complete successfully

### Actual Behaviour

The build throws [vite:build-import-analysis] parse error. This seems to happen when importing from outside the src directory, but I have also had this error occur from imports inside the src. I'm having trouble consistently reproducing the issue from inside the src but I suspect they are related

### Additional Information

https://stackblitz.com/edit/qwik-starter-e1wsbm?file=package.json
Run `npm run build` in the console

### PR DESCRIPTION
**Pull request by zanettin**
# What is it?

- [ ] Feature / enhancement
- [ ] Bug
- [x] Docs / tests

# Description

Added docs for vite within the advanced section covering configuration options for the `qwikVite()` and `qwikCity()` plugins along with a short troubleshooting block.

fixes #1885 (https://github.com/BuilderIO/qwik/issues/1885#issuecomment-1420022458)

# Checklist:

- [x] My code follows the [developer guidelines of this project](https://github.com/BuilderIO/qwik/blob/main/CONTRIBUTING.md)
- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [ ] Added new tests to cover the fix / functionality

### CODE REVIEW

File: packages/docs/src/routes/docs/overview/index.mdx [hunk 1]
```diff
@@ -58,8 +58,7 @@ Qwik is a new kind of web framework that can deliver instant loading web applica
   </div>
   <div class="card">
     <h3>Optimized for speed</h3>
-    <p>
-      Qwik has unprecedented performance, offering sub-second full page loads even on mobile
+    <p>Qwik has unprecedented performance, offering sub-second full page loads even on mobile
```
Comment (zanettin) [inline packages/docs/src/routes/docs/overview/index.mdx:61]: with a line break after the `<p>` tag the following error was thrown when running the docs locally:
```
10:07:06 PM [vite] Internal server error: Expected a closing tag for `<p>` (61:5-61:8) before the end of `paragraph`
  Plugin: vite-plugin-qwik-city
  File: /Users/zanettin/development/qwik/qwik/packages/docs/src/routes/docs/overview/index.mdx:60:5
  58 |    </div>
  59 |    <div class="card">
  60 |      <h3>Optimized for speed</h3>
     |       ^
  61 |      <p>Qwik has unprecedented performance, offering sub-second full page loads even on mobile
  62 |        devices. Qwik achieves this by delivering pure HTML, and incrementally loading JS only
```

File: packages/docs/src/routes/docs/advanced/vite/index.mdx [hunk 2]
```diff
@@ -0,0 +1,219 @@
...
+     * If set to `false`, qwik dev inspector will be disabled.
+     * Valid values are `KeyboardEvent.code` values.
+     * Please note that the 'Left' and 'Right' suffixes are ignored.
+     */
+    clickToSource: string[] | false;
+};
+```
+
+### `qwikCity()`
+To modify the configuration, you can pass an object to the `qwikVite` function. Possible options are:
```
Comment (literalpie) [inline packages/docs/src/routes/docs/advanced/vite/index.mdx:165]:
```suggestion
To modify the configuration, you can pass an object to the `qwikCity` function. Possible options are:
```

Comment (zanettin) [reply packages/docs/src/routes/docs/advanced/vite/index.mdx]: nice catch 🙏 thanks a lot!

File: packages/docs/src/routes/docs/overview/index.mdx (hunk 1 above)
Comment (manucorporat) [reply packages/docs/src/routes/docs/overview/index.mdx]: i saw it before, i am not sure. seems like mdx is generating nested <p> for some reason

**Comment by zanettin** [comment]
@manucorporat @shairez do you think this is valuable for the community and we can merge it or should we invest more to make the API documentation available for all on the docs?

**Comment by manucorporat** [comment]
I would merge, we cant write the perfect docs and perfect organization at once. Lets code, ship, test, refactor, repeat

**Pull request merged**